File: Front/src/app/models/intershipoffer.ts
//import { Postulation } from "./postulation.model";

export enum TypeInternship {
  STAGE_FORMATION_HUMAINE_SOCIALE = '  INTERNSHIP_FORMATION_HUMAINE_ET_SOCIALE',
  STAGE_IMMERSION_ENTREPRISE = 'INTERNSHIP_D_IMMERSION_EN_ENTREPRISE',
  STAGE_INGENIEUR = 'INTERNSHIP_INGENIEUR',
  STAGE_PFE = 'INTERNSHIP_PFE' ,


}
export class intershipoffer {
  idsujet!: number;
  title!: string;
  description!: string;
  location!: string;
   duration!: number;
  numberOfStudents!: number;
  requirements!: string;
  companyname!: string;
  companymail!: string;
  typeInternship!: TypeInternship;
  creationDate!: string;


  imageUrl!: string;

  
 // postulationtSet!: Postulation[];
}

File: Back/src/main/java/com/example/back/entities/IntershipOffer.java
package com.example.back.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class IntershipOffer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idsujet;
    private String title;
    private String description ;
    private int duration ;
    private String location;
    private String requirements;
    private int numberOfStudents;
    private String imageUrl;

    @Temporal(TemporalType.TIMESTAMP)
    private Date creationDate;

    @Enumerated(EnumType.STRING)
    private TypeInternship typeInternship ;

    private Long idcompany;




    @OneToMany(cascade = CascadeType.ALL, mappedBy="intershipOffer")
    private Set<Postulation> postulations ;




}
